src/coffee/export.coffee

```diff
@@ -230,16 +232,22 @@ class Export
         )
         # remove all the variants that don't meet the price condition
         product.variants = _.compact(product.variants)
-        csv = csv.concat exportMapper.mapProduct(
+        data = data.concat exportMapper.mapProduct(
```

Comment:
Every `exportMapper.mapProduct` will return an array of masterVariant and possible variants. Then all these rows are added to one big buffer variable `data` which is then flushed using `write` function. 

This version which you are proposing would have to write products separately one by one as the `data` variable would change every time we map new product to CSV rows